sources/api/certdog/src/main.rs

```diff
@@ -0,0 +1,355 @@
+        },
+
+        #[snafu(display("Failed to parse cert: {}", source))]
+        ParseX509Certificate {
+            source: x509_parser::nom::Err<x509_parser::error::X509Error>,
+        },
+
+        #[snafu(display("Error while reading bundle from file {}: '{}'", path.display(), source))]
+        ReadSourceBundle { path: PathBuf, source: io::Error },
+
+        #[snafu(display("Error deserializing response from '{}': '{}'", uri, source))]
+        ResponseJson {
+            uri: String,
+            source: serde_json::Error,
+        },
+
+        #[snafu(display("Failed to update trust store: {}", source))]
+        UpdateTrustedStore { source: io::Error },
+
+        #[snafu(display("Failed to write to pem string: {}", source))]
+        WritePEMString { source: std::fmt::Error },
+    }
+}
+
+type Result<T> = std::result::Result<T, error::Error>;
+
+#[cfg(test)]
+mod test_certdog {
+    use model;
+    use model::modeled_types::{Identifier, PemCertificateString};
+    use std::collections::HashMap;
+    use std::convert::TryFrom;
+    use std::fs::File;
+
+    static TEST_PEM: &str = include_str!("../../../models/tests/data/test-pem");
+
+    #[test]
+    fn split_bundles() {
+        let mut bundle = HashMap::new();
+        bundle.insert(
+            Identifier::try_from("trusted").unwrap(),
+            model::PemCertificate {
+                data: Some(PemCertificateString::try_from(TEST_PEM).unwrap()),
+                trusted: Some(true),
+            },
+        );
+        bundle.insert(
+            Identifier::try_from("distrusted").unwrap(),
+            model::PemCertificate {
+                data: Some(PemCertificateString::try_from(TEST_PEM).unwrap()),
+                trusted: Some(false),
+            },
+        );
+        bundle.insert(
+            Identifier::try_from("distrusted-without-flag").unwrap(),
+            model::PemCertificate {
+                data: Some(PemCertificateString::try_from(TEST_PEM).unwrap()),
+                trusted: None,
+            },
+        );
+
+        let splitted = super::split_bundles(bundle).unwrap();
+        // The test-pem file contains two X509 certificates
+        assert!(splitted.trusted_certs.len() == 2);
+        assert!(splitted.distrusted_certs.len() == 4);
+    }
+
+    #[test]
+    fn update_trusted_store() {
+        let trusted_store = tempfile::NamedTempFile::new().unwrap();
+        let source_bundle = tempfile::NamedTempFile::new().unwrap();
+        let (_, pem) =
+            x509_parser::pem::parse_x509_pem(&base64::decode(TEST_PEM.as_bytes()).unwrap())
+                .unwrap();
+        let mut trusted_certs: Vec<x509_parser::pem::Pem> = Vec::new();
+        trusted_certs.push(pem);
+        let certs_bundle = super::CertBundle {
+            trusted_certs,
+            distrusted_certs: Vec::new(),
+        };
+        assert!(super::update_trusted_store(certs_bundle, &trusted_store, &source_bundle).is_ok());
```

Comment:
It's common to add `use super::*` at the top of `mod test_certdog`

```suggestion
        assert!(update_trusted_store(certs_bundle, &trusted_store, &source_bundle).is_ok());
```